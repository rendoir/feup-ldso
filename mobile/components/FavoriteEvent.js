import React from 'react';
import {
    StyleSheet,
    Image
} from 'react-native';

import { Card, Icon, View, Text } from 'native-base';

export default class FavoriteEvent extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            imageLoaded: true
        };

        this.onFavoriteCall = this.onFavoriteCall.bind(this);

    }

    ImageLoadingError() {
        this.setState({ imageLoaded: false });
    }

    getMonthInString(month) {
        switch (parseInt(month)) {
        case 1: return 'Janeiro';
        case 2: return 'Fevereiro';
        case 3: return 'Março';
        case 4: return 'Abril';
        case 5: return 'Maio';
        case 6: return 'Junho';
        case 7: return 'Julho';
        case 8: return 'Agosto';
        case 9: return 'Setembro';
        case 10: return 'Outubro';
        case 11: return 'Novembro';
        case 12: return 'Dezembro';
        default: return 'Error';
        }
    }

    getDateinString() {
        let day = this.props.start_date.split('T')[0].split('-')[2];
        let month = this.getMonthInString(this.props.start_date.split('T')[0].split('-')[1]);
        let time = this.props.start_date.split('T')[1].split(':')[0] + ':' + this.props.start_date.split('T')[1].split(':')[1];
        let date = day + ' de ' + month + ' - ' + time;
        return date;
    }

    getPrice() {
        return (this.props.price == 0) ? 'Gratuito' : (this.props.price + '€');
    }

    onFavoriteCall() {
        this.props.onFavorite(this.props.id);
    }

    render() {
        return (
            <Card onPress={this.props.onPress} style={{ backgroundColor: 'white', height: 125 }}>

                <View style={{ flexDirection: 'row', height: '100%' }}>
                    <View style={{ flex: 2 }}>
                        <Image source={this.state.imageLoaded ? { uri: 'http://' + global.api + ':3030/' + this.props.id } : require('../assets/images/default.png')}
                            style={styles.image}
                            onError={this.ImageLoadingError.bind(this)}
                            onPress={this.props.onPress} />
                    </View>
                    <View style={{ flex: 5, borderLeftColor: '#D05722', borderLeftWidth: 5, justifyContent: 'space-between', paddingHorizontal: '2%', paddingBottom: '2%' }}>
                        <Text style={{ color: 'black', fontFamily: 'OpenSans-Regular', fontSize: 20, fontWeight: 'bold' }} numberOfLines={1} onPress={this.props.onPress}>{this.props.title}</Text>
                        <Text style={{ color: 'black', fontFamily: 'OpenSans-Regular', fontSize: 18 }} numberOfLines={1} onPress={this.props.onPress}>{this.getDateinString()}</Text>
                        <View style={{ flexDirection: 'row', justifyContent: 'space-between' }} >
                            <Text style={{ color: 'black', fontFamily: 'OpenSans-Regular', fontSize: 18 }} numberOfLines={1} onPress={this.props.onPress}>Preço: {this.getPrice()}</Text>
                            <Icon className="fave_icon" style={{ fontSize: 25, alignSelf: 'center', color: '#FF5722' }} type='FontAwesome' name={this.props.is_favorite ? 'heart' : 'heart-o'} onPress={this.onFavoriteCall} />
                        </View>
                    </View>
                </View>

                {/* <Image source={this.state.imageLoaded ? { uri: 'http://' + global.api + ':3030/' + this.props.id } : require('../assets/images/default.png')}
                    style={styles.image}
                    onError={this.ImageLoadingError.bind(this)}
                    onPress={this.props.onPress} />
                <View style={{ flexDirection: 'row', backgroundColor: 'white' }} onPress={this.props.onPress}>
                    <Badge style={{ flex: 1, backgroundColor: '#002040', margin: '3%', height: 68, alignItems: 'center' }}>
                        <Text style={{ color: 'white', fontFamily: 'OpenSans-Regular', fontSize: 18, lineHeight: 30 }} onPress={this.props.onPress}>{this.props.start_date.split('T')[0].split('-')[2]}</Text>
                        <Text style={{ color: 'white', fontFamily: 'OpenSans-Regular', fontSize: 18, lineHeight: 30 }} onPress={this.props.onPress}>{this.getMonthInString(this.props.start_date.split('T')[0].split('-')[1])}</Text>
                    </Badge>
                    <View style={{ flex: 5, marginRight: '3%', marginVertical: '3%' }} onPress={this.props.onPress}>
                        <Text style={{ color: 'black', fontFamily: 'OpenSans-Regular', fontSize: 18, fontWeight: 'bold' }} numberOfLines={1} onPress={this.props.onPress}>{this.props.title}</Text>
                        <Text style={{ color: 'black', fontFamily: 'OpenSans-Regular', fontSize: 16 }} numberOfLines={1} onPress={this.props.onPress}>{this.props.location} - {this.props.start_date.split('T')[1].split(':')[0] + ':' + this.props.start_date.split('T')[1].split(':')[1]}</Text>
                        <Text style={{ color: 'black', fontFamily: 'OpenSans-Regular', fontSize: 16 }} numberOfLines={1} onPress={this.props.onPress}>Preço: {this.props.price}€</Text>
                    </View>
                    <Icon className="fave_icon" style={{ fontSize: 35, flex: 1, alignSelf: 'center', color: '#D05722' }} type='FontAwesome' name={this.props.is_favorite ? 'heart' : 'heart-o'} onPress={this.onFavoriteCall} />
        </View> */}
            </Card>
        );
    }
}

const styles = StyleSheet.create({
    image: {
        width: null,
        flex: 1
    }
});
