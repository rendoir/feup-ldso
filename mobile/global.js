global.api = "188.166.95.23";
global.userId = null;
global.userName = null;

global.dictionary = {
    FAVORITES: { PT: "Favoritos", EN: "Favorites" },
    AGENDA: { PT: "Agenda", EN: "Agenda" },
    SEARCH: { PT: "Pesquisa", EN: "Search" },
    EVENTS: { PT: "Eventos", EN: "Events" },
    CATEGORY: { PT: "Categoria", EN: "Category" },
    CATEGORIES: { PT: "Categorias", EN: "Categories" },
    ENTITY: { PT: "Entidade", EN: "Entity" },
    ENTITIES: { PT: "Entidades", EN: "Entities" },
    ALL: { PT: "Todos", EN: "All" },
    NO_EVENTS: { PT: "Não há eventos futuros para mostrar.", EN: "There are no future events to display." },
    AT: { PT: "em", EN: "at" },
    PRICE: { PT: "Preço", EN: "Price" },
    FREE: { PT: "Gratuito", EN: "Free" },
    MAP: { PT: "Mapa", EN: "Map" },
    CALENDAR: { PT: "Calendário", EN: "Calendar" },
    APP_NAME: { PT: "Agenda Cultural da Universidade do Porto", EN: "Porto University's Cultural Agenda" },
    GOOGLE_SIGNIN: { PT: "Entrar com conta Google", EN: "Sign in with Google account" },
    NO_SEARCH_RESULTS: { PT: "Não há eventos para a pesquisa feita.", EN: "There are no events that match your search." },
    NO_FAVORITES: { PT: "Não há eventos favoritos para mostrar.", EN: "There are no favorite events." }
};

global.calendar = {
    1: { PT: 'JAN', EN: 'JAN' },
    2: { PT: 'FEV', EN: 'FEB' },
    3: { PT: 'MAR', EN: 'MAR' },
    4: { PT: 'ABR', EN: 'APR' },
    5: { PT: 'MAI', EN: 'MAY' },
    6: { PT: 'JUN', EN: 'JUN' },
    7: { PT: 'JUL', EN: 'JUL' },
    8: { PT: 'AGO', EN: 'AUG' },
    9: { PT: 'SET', EN: 'SEP' },
    10: { PT: 'OUT', EN: 'OCT' },
    11: { PT: 'NOV', EN: 'NOV' },
    12: { PT: 'DEZ', EN: 'DEC' }
};

global.fullCalendar = {
    1: { PT: 'Janeiro', EN: 'January' },
    2: { PT: 'Fevereiro', EN: 'February' },
    3: { PT: 'Março', EN: 'March' },
    4: { PT: 'Abril', EN: 'April' },
    5: { PT: 'Maio', EN: 'May' },
    6: { PT: 'Junho', EN: 'June' },
    7: { PT: 'Julho', EN: 'July' },
    8: { PT: 'Agosto', EN: 'August' },
    9: { PT: 'Setembro', EN: 'September' },
    10: { PT: 'Outubro', EN: 'October' },
    11: { PT: 'Novembro', EN: 'November' },
    12: { PT: 'Dezembro', EN: 'December' }
};
