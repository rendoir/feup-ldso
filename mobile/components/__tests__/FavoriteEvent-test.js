/**
 * @jest-environment jsdom
 */

import 'react-native';
import React from 'react';
import FavoriteEvent from '../FavoriteEvent';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '../../global.js';

Enzyme.configure({ adapter: new Adapter() });


describe('Event Test', () => {

    it('renders correctly', () => {
        const eventData = { "id": 2, "title": "Feup Career Fair 2018", "description": "Nos dias 16, 17, 18 e 19 de outubro acontece mais uma edição da FEUP CAREER FAIR e a FEUP convida todos os estudantes e graduados a estarem presentes!", "start_date": "2018-10-15T23:00:00.000Z", "end_date": "2018-10-18T23:00:00.000Z", "location": "FEUP", "price": 0, "entity_id": 9, "user_id": 2, "is_favorite": true };
        const tree = renderer.create(<FavoriteEvent {...eventData} language='PT' />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly in english', () => {
        const eventData = { "id": 2, "title": "Feup Career Fair 2018", "title_english": "Feup Career Fair 2018", "description": "Nos dias 16, 17, 18 e 19 de outubro acontece mais uma edição da FEUP CAREER FAIR e a FEUP convida todos os estudantes e graduados a estarem presentes!", "start_date": "2018-10-15T23:00:00.000Z", "end_date": "2018-10-18T23:00:00.000Z", "location": "FEUP", "price": 0, "entity_id": 9, "user_id": 2, "is_favorite": true };
        const tree = renderer.create(<FavoriteEvent {...eventData} language='EN' />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('getMonthInString returns correctly', () => {
        const eventData = { "id": 2, "title": "Feup Career Fair 2018", "description": "Nos dias 16, 17, 18 e 19 de outubro acontece mais uma edição da FEUP CAREER FAIR e a FEUP convida todos os estudantes e graduados a estarem presentes!", "start_date": "2018-10-15T23:00:00.000Z", "end_date": "2018-10-18T23:00:00.000Z", "location": "FEUP", "price": 1, "entity_id": 9, "user_id": 2, "is_favorite": true };
        const tree = renderer.create(<FavoriteEvent {...eventData} language='PT' />).getInstance();

        expect(tree.getMonthInString('01')).toEqual("Janeiro");
        expect(tree.getMonthInString('02')).toEqual("Fevereiro");
        expect(tree.getMonthInString('03')).toEqual("Março");
        expect(tree.getMonthInString('04')).toEqual("Abril");
        expect(tree.getMonthInString('05')).toEqual("Maio");
        expect(tree.getMonthInString('06')).toEqual("Junho");
        expect(tree.getMonthInString('07')).toEqual("Julho");
        expect(tree.getMonthInString('08')).toEqual("Agosto");
        expect(tree.getMonthInString('09')).toEqual("Setembro");
        expect(tree.getMonthInString('10')).toEqual("Outubro");
        expect(tree.getMonthInString('11')).toEqual("Novembro");
        expect(tree.getMonthInString('12')).toEqual("Dezembro");
    });

    it('image loading', () => {
        const eventData = { "id": 2, "title": "Feup Career Fair 2018", "description": "Nos dias 16, 17, 18 e 19 de outubro acontece mais uma edição da FEUP CAREER FAIR e a FEUP convida todos os estudantes e graduados a estarem presentes!", "start_date": "2018-10-15T23:00:00.000Z", "end_date": "2018-10-18T23:00:00.000Z", "location": "FEUP", "price": 0, "entity_id": 9, "user_id": 2, "is_favorite": true };
        const tree = renderer.create(<FavoriteEvent {...eventData} language='PT' />).getInstance();

        tree.ImageLoadingError();
        expect(tree.state.imageLoaded).toEqual(false);
    });


    it('favorite an event works', (done) => {

        const eventData = {
            "id": 2,
            "title": "Feup Career Fair 2018",
            "description": "Nos dias 16, 17, 18 e 19 de outubro acontece mais uma edição da FEUP CAREER FAIR e a FEUP convida todos os estudantes e graduados a estarem presentes!",
            "start_date": "2018-10-15T23:00:00.000Z",
            "end_date": "2018-10-18T23:00:00.000Z",
            "location": "FEUP",
            "price": 0,
            "entity_id": 9,
            "user_id": 2,
            "is_favorite": false
        };

        let mock = jest.fn();
        const wrapper = shallow(<FavoriteEvent {...eventData} onFavorite={mock} language='PT' />);

        wrapper.instance().onFavoriteCall();

        setImmediate(() => {
            expect(mock).toHaveBeenCalled();
            done();
        });
    });

});
