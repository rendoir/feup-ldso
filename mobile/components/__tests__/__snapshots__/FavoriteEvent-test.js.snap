// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "backgroundColor": "white",
        "height": 125,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flexDirection": "row",
          "height": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 2,
          },
        ]
      }
    >
      <Image
        onError={[Function]}
        source={
          Object {
            "uri": "http://188.166.95.23:3030/2",
          }
        }
        style={
          Object {
            "flex": 1,
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "borderLeftColor": "#D05722",
            "borderLeftWidth": 5,
            "flex": 5,
            "justifyContent": "space-between",
            "paddingBottom": "2%",
            "paddingHorizontal": "2%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
      >
        Feup Career Fair 2018
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 18,
            },
          ]
        }
        uppercase={false}
      >
        15 de Outubro às 23:00
      </Text>
      <View
        style={
          Array [
            Object {},
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "black",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 18,
              },
            ]
          }
          uppercase={false}
        >
          Preço
          : 
          Gratuito
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Event Test renders correctly in english 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "backgroundColor": "white",
        "height": 125,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flexDirection": "row",
          "height": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 2,
          },
        ]
      }
    >
      <Image
        onError={[Function]}
        source={
          Object {
            "uri": "http://188.166.95.23:3030/2",
          }
        }
        style={
          Object {
            "flex": 1,
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "borderLeftColor": "#D05722",
            "borderLeftWidth": 5,
            "flex": 5,
            "justifyContent": "space-between",
            "paddingBottom": "2%",
            "paddingHorizontal": "2%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
      >
        Feup Career Fair 2018
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 18,
            },
          ]
        }
        uppercase={false}
      >
        October 15 at 23:00
      </Text>
      <View
        style={
          Array [
            Object {},
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "black",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 18,
              },
            ]
          }
          uppercase={false}
        >
          Price
          : 
          Free
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;
